### DISCUSSION TITLE
How does actions update the store?

### DISCUSSION THREAD

**Discussion by khmob** [vuejs/pinia#1888] [Help and Questions] [answered]
Both of these give the same result, `answeredObject` which is part of the state gets updated:

```
this.answeredObject.map(review => {
                if (this.objects[name]) {
                    object = {
                        key: this.objects[name],
                        time: Date.now().toString(),
                    }
                }
                return review
            })
```

and 

```
this.answeredObject = this.answeredObject.map(review => {
                if (this.objects[name]) {
                    object = {
                        key: this.objects[name],
                        time: Date.now().toString(),
                    }
                }
                return review
            })
```

How does that work? In general, `map` doesn't update the current array, it creates a new one. So how does pinia do that?

**Comment by BenShelton** [comment, marked as answer]
Pinia doesn't do anything special with `map`. Your `map` isn't actually doing anything to the array, it just returns the same object. It looks like the assignment to the `object` variable (`object = `) is what is manipulating something in both cases, and that is probably the update you are actually seeing. This is known as a _side effect_, where the function is updating something outside of it.
